libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -298,19 +298,22 @@ method onPubSubPeerEvent*(p: PubSub, peer: PubSubPeer, event: PubSubPeerEvent) {
 method getOrCreatePeer*(
     p: PubSub,
     peerId: PeerId,
-    protos: seq[string]): PubSubPeer {.base, gcsafe.} =
+    protosToDial: seq[string],
+    protoNegotiated: string = ""): PubSubPeer {.base, gcsafe.} =
 
   p.peers.withValue(peerId, peer):
+    if peer[].codec == "" and protoNegotiated != "":
```

Comment:
This solution is ugly, but couldn't think of anything better. Ideas are welcome.

Reply:
`if peer[].codec == "":` will have the same result in the end. And it's a bit better, I guess?

Reply:
True